tests/perf/test_accuracy.py

```diff
@@ -2,142 +2,255 @@
-            self.anns_to_gt_bboxes = voc_anns_to_gt_bboxes
+        for lrange in range(0, max_num, batch_size):
+            rrange = min(max_num, lrange + batch_size)
+
+            start_time = time()
+            images, explanations, dataset_gt_bboxes = [], [], []
+            for i in range(lrange, rrange):
+                image, anns = self.dataset[i]
+                image_np = np.array(image)  # PIL -> np.array
+                gt_bbox_dict = self.anns_to_gt_bboxes(anns, self.dataset_labels_dict)
 
-    def test_explainer_images(self):
-        images, explanations, dataset_gt_bboxes = [], [], []
-        for image, anns in self.dataset:
-            image_np = np.array(image)
-            gt_bbox_dict = self.anns_to_gt_bboxes(anns, self.dataset_labels_dict)
-            targets = [target for target in gt_bbox_dict.keys() if target in VOC_NAMES]
+                # To measure the quality of predicted saliency maps without the gt info from the dataset (found out how to check it)
+                # targets = np.argmax(self.model_predict(image_np))
+                targets = list(gt_bbox_dict.keys())
+                intersected_targets = list(set(targets) & set(self.dataset_label_list))
+                if len(intersected_targets) == 0:
+                    # Skip images where gt classes and model classes do not match
+                    continue
+                explanation = self.explainer(
+                    image_np,
+                    targets=intersected_targets,
+                    label_names=self.dataset_label_list,
+                    colormap=False,
+                    **black_box_kwargs,
+                )
+                images.append(image_np)
+                explanations.append(explanation)
+                dataset_gt_bboxes.append(gt_bbox_dict)
+
+            # Write per-batch statistics to track failures
+            explained_images += len(explanations)
+            record = {"range": f"{lrange}-{rrange}"}
+            record.update(self.get_scores_times(explanations, images, dataset_gt_bboxes, start_time))
+            records.append(record)
+
+            df = pd.DataFrame([record]).round(3)
+            df.to_csv(self.data_metric_path / f"accuracy_{self.dataset_name}.csv", mode="a", header=False, index=False)
+
+        experiment_time = time() - experiment_start_time
+        mean_scores_dict = {"explained_images": explained_images, "overall_time": experiment_time}
+        mean_scores_dict.update(
+            {
+                key: np.mean([record[key] for record in records if key in record])
+                for key in records[0].keys()
+                if key != "range"
+            }
+        )
+        df = pd.DataFrame([mean_scores_dict]).round(3)
+        df.to_csv(self.data_metric_path / f"mean_accuracy_{self.dataset_name}.csv", index=False)
 
-            explanation = self.explainer(image_np, targets=targets, label_names=VOC_NAMES, colormap=False)
+    def get_scores_times(
```

Comment:
Do we really require `time`? This is the time of metric measurement, right? I would propose to eliminate it. Main intention of this test is to measure metrics. Sure, it is good to do it fast, but it is secondary. 
`get_scores_times` -> `get_scores`
Also I would propose to eliminate time from the final record table (if it is there). Although I am ok to have an auxiliary table with time of measurement, for a reference.

Reply:
The timing information was for internal debugging, so I'll rename the function. Thank you for the suggestion.

Regarding time reporting, I agree, it can be tricky for the legal reasons, but it's useful to see how time-consuming different explanation methods are and how long it takes to measure quality. What if we keep the timing information but report it in `model inferences` instead of `hours and minutes`? We have this value for all the explanation methods and metrics.

Or leave only accuracy metrics and explain time (in model inferences) for `Methods` table and create table for `Metrics` with comparison of consumed time (in model inferences) for each metric for each method.
@goodsong81 What do you think about it?

Reply:
If we could have accuracy & efficiency comparison for the methods, any options are all OK to me.